analyzers/src/SonarAnalyzer.Common/DiagnosticAnalyzer/DiagnosticFactory.cs

```diff
@@ -0,0 +1,31 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace SonarAnalyzer.Helpers;
+
+public static class DiagnosticFactory
+{
+    public static Diagnostic Create(DiagnosticDescriptor descriptor, Compilation compilation, Location location, IEnumerable<Location> additionalLocations, params object[] messageArgs) =>
```

Comment:
Can we make it an extension method?
```suggestion
    public static Diagnostic CreateDiagnostic(this DiagnosticDescriptor descriptor, Compilation compilation, Location location, IEnumerable<Location> additionalLocations, params object[] messageArgs) =>
```